target/scripts/startup/setup-stack.sh

```diff
@@ -67,20 +67,22 @@ function _setup_default_vars
   PFLOGSUMM_RECIPIENT="${PFLOGSUMM_RECIPIENT:=${REPORT_RECIPIENT}}"
   LOGWATCH_RECIPIENT="${LOGWATCH_RECIPIENT:=${REPORT_RECIPIENT}}"
 
+  VARS[LOGWATCH_RECIPIENT]="${LOGWATCH_RECIPIENT}"
+  VARS[PFLOGSUMM_RECIPIENT]="${PFLOGSUMM_RECIPIENT}"
+  VARS[PFLOGSUMM_SENDER]="${PFLOGSUMM_SENDER}"
+  VARS[PFLOGSUMM_TRIGGER]="${PFLOGSUMM_TRIGGER}"
+  VARS[POSTMASTER_ADDRESS]="${POSTMASTER_ADDRESS}"
+  VARS[REPORT_RECIPIENT]="${REPORT_RECIPIENT}"
+  VARS[REPORT_SENDER]="${REPORT_SENDER}"
+
+  : >/root/.bashrc
+  : >/etc/dms-settings
   local VAR
   for VAR in "${!VARS[@]}"
   do
     echo "export ${VAR}='${VARS[${VAR}]}'" >>/root/.bashrc
+    echo "${VAR}='${VARS[${VAR}]}'"        >>/etc/dms-settings
```

Comment:
I didn't see in the discussion what the point of dropping the `export` prefix was? Couldn't you just dump these to `/etc/dms-settings` as before instead of `/root/.bashrc`? Then have `/root/.bashrc` source `/etc/dms-settings`?

Or was there some reason to want to temporarily set/override ENV without making it available to subprocesses?

I'm fine with whatever you go with, you have a better understanding of this stuff than me :sweat_smile: It seems better than maintaining a bunch of conditional default assignments like the previous approach :+1: (_except for loss of compatibility with `docker run` or the `docker exec ... sh -c ...` overrride_)

Reply:
The reason is pretty simple: The file should only contain the current container settings and not any further commands/logic. So if a script needs them (afaik this is the first time), the file can just be sourced. There is no need for "export". If a script needs **our** variables exported to a subprocess, it can do it on its own.

Reply:
Just seemed more consistent in behaviour/expectation if we kept the `export` as if `.bashrc` isn't loaded (eg: because of `bash -c`), then we're still able to make the expected environment available to our scripts. And I say that in the sense that sometimes calling a command may have the environment already made available to it, while the other context is sourcing it without the `export`.

I think we're more likely to run into troubleshooting issues from that inconsistency more than we would be from issues of having the env exported when it's one of the expected contexts and the whole point of this PR is a workaround for when that hasn't happened to ensure defaults are correctly fallen back to.

---

Not my forte here, so if your more confident in avoiding `export` as the better approach go for it :sunglasses:

Reply:
> as if .bashrc isn't loaded

For clarification: the DMS variable stuff in .bashrc is not used in any way. It's just there (I think), to quickly check what ENVs the running container uses. So when you start a bash session, you can easily do something like `echo $SOME_DMS_VAR`.
That said, I don't think the exports in .bashrc aren't needed for anything DMS rely on.

The only script that need to access a DMS var (ENABLE_QUOTA) is `listmailuser`. I don't think introducing a entrypoint script is worth it. If there is more demand/scripts in the future, we can still implement it.

Reply:
The whole .bashrc thing started with [this](https://github.com/docker-mailserver/docker-mailserver/commit/08dc28e3041550646cf8b45ccfccada2a0ffde3c) for the `VIRUSMAILS_DELETE_DELAY`.

Reply:
> start-mailserver.sh: default to 7 (makes the Dockerfile ENV a bit redundant? When the Dockerfile ENV was added in 2017 this var was removed, but then added back in 2018 to resolve an issue)

I think there is a room for improvements 😉 I am wondering, if any of the ENVs in Dockerfile nowadays are needed.

Reply:
>  Edit: The only exception is VIRUSMAILS_DELETE_DELAY.

Or maybe not 😆 You also wrote, that `VIRUSMAILS_DELETE_DELAY` is also in `/etc/environment`.